packages/app-contracts/src/Codes/Add.tsx

```diff
@@ -0,0 +1,118 @@
+// Copyright 2017-2019 @polkadot/app-contracts authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import React from 'react';
+import { Button, Input } from '@polkadot/ui-app';
+import { Hash } from '@polkadot/types';
+
+import ContractModal, { ContractModalProps, ContractModalState } from '../Modal';
+
+import ValidateCode from './ValidateCode';
+import store from '../store';
+import translate from '../translate';
+
+type Props = ContractModalProps;
+
+type State = ContractModalState & {
+  codeHash: string,
+  isBusy: boolean,
+  isCodeValid: boolean
+};
+
+class Add extends ContractModal<Props, State> {
+  constructor (props: Props) {
+    super(props);
+    this.defaultState = {
+      ...this.defaultState,
+      codeHash: '',
+      isBusy: false,
+      isCodeValid: false
+    };
+    this.state = this.defaultState;
```

Comment:
Define `defaultState` on the component, i.e. 

```js
class Add extends ContractModal<Props, State> {
  state: State;
  defaultState: State;
```

(That way, when used, or not initialised, TS complains)